MapboxCoreNavigation/RouteController.swift

```diff
@@ -162,7 +164,48 @@ extension RouteController: CLLocationManagerDelegate {
         let radius = min(RouteControllerMaximumDistanceBeforeRecalculating,
                          location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
 
-        return newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
+        let isCloseToCurrentStep = newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
+        
+        // If the user is moving away from the maneuver location
+        // and they are close to the next step
+        // we can safely say they have completed the maneuver.
+        // This is intended to be a fallback case when we do find
+        // that the users course matches the exit bearing.
+        if let upComingStep = routeProgress.currentLegProgress.upComingStep {
+            let isCloseToUpComingStep = newLocation.isWithin(radius, of: upComingStep)
+            if !isCloseToCurrentStep && isCloseToUpComingStep && isInsideManeuverRadius {
```

Comment:
That's great, but as I mentioned above, what if the device is unable to update its location while within the maneuver zone but the user has nonetheless proceeded to the next step? (Maybe they passed through the intersection quickly.) What if the intersection happens to be larger than the maneuver zone? (We should probably have different maneuver zone radii for different types of maneuvers.)